### PR TITLE
Handle deletion for comments and threads in realtime

### LINKED ISSUE #6586: [Bug] Comment and thread deletion doesn't happen in realtime for concurrent users
## Description
Comment and thread deletion doesn't happen in realtime for concurrent users

### Steps to reproduce the behaviour:
1. Create an app and invite another user
2. Open the editor as the other user
3. Add a couple of comments on a new thread
4. Delete a comment: observe the comment doesn't get deleted for the other user
5. Delete a thread: observe the thread doesn't get deleted for the other user

### PR DESCRIPTION
**Pull request by rishabhsaxena**
## Description
Handle deletion for comments and threads in realtime

Fixes #6586

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
- manually

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: handle-delete-thread-and-comment-events 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 54.8 **(-0.02)** | 36.74 **(-0.02)** | 33.11 **(-0.04)** | 55.38 **(-0.01)**
 :red_circle: | app/client/src/actions/commentActions.ts | 80.15 **(-0.65)** | 100 **(0)** | 45.24 **(-2.26)** | 78.46 **(0.68)**
 :green_circle: | app/client/src/reducers/uiReducers/commentsReducer/commentsReducer.ts | 54.41 **(8.46)** | 50 **(20)** | 55.56 **(0)** | 61.67 **(9.36)**
 :sparkles: :new: | **app/client/src/reducers/uiReducers/commentsReducer/common.ts** | **17.39** | **0** | **0** | **11.11**
 :sparkles: :new: | **app/client/src/reducers/uiReducers/commentsReducer/handleDeleteCommentEvent.ts** | **37.5** | **100** | **0** | **50**
 :sparkles: :new: | **app/client/src/reducers/uiReducers/commentsReducer/handleDeleteCommentThreadEvent.ts** | **37.5** | **100** | **0** | **50**
 :red_circle: | app/client/src/sagas/WebsocketSagas/handleSocketEvent.ts | 17.86 **(-2.14)** | 0 **(0)** | 0 **(0)** | 18.87 **(-2.41)**</details>

### CODE REVIEW

File: app/client/src/reducers/uiReducers/commentsReducer/common.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  return state;
+};
+
+export const deleteCommentFromState = (
+  state: CommentsReduxState,
+  commentId: string,
+  threadId: string,
+) => {
+  const commentThread = state.commentThreadsMap[threadId];
+  state.commentThreadsMap[threadId] = {
```
Comment (pranavkanade) [inline app/client/src/reducers/uiReducers/commentsReducer/common.ts:38]: Should consider the possibility of commentThread being `undefined`
```
  if(!commentThread) return state;
```

File: app/client/src/reducers/uiReducers/commentsReducer/commentsReducer.ts [hunk 2]
```diff
@@ -216,26 +213,15 @@ const commentsReducer = createReducer(initialState, {
...
-
-    delete state.commentThreadsMap[commentThreadId];
-
-    state.commentThreadsMap = { ...state.commentThreadsMap };
+    if (!state.applicationCommentThreadsByRef[appId]) return state;
 
-    state.applicationCommentThreadsByRef[appId as string] = {
-      ...state.applicationCommentThreadsByRef[appId as string],
-    };
+    const updatedState = deleteCommentThreadFromState(
```
Comment (pranavkanade) [inline app/client/src/reducers/uiReducers/commentsReducer/commentsReducer.ts:218]: Can't we reuse `handleDeleteCommentThreadEvent` functions as well? or have same piece of code run for both `DELETE_COMMENT_THREAD_EVENT ` and `DELETE_THREAD_SUCCESS`?

File: app/client/src/reducers/uiReducers/commentsReducer/commentsReducer.ts [hunk 3]
```diff
@@ -143,15 +146,9 @@ const commentsReducer = createReducer(initialState, {
...
     const { commentId, threadId } = action.payload;
 
-    const commentThread = state.commentThreadsMap[threadId];
-    state.commentThreadsMap[threadId] = {
-      ...commentThread,
-      comments: commentThread.comments.filter(
-        (comment) => comment.id !== commentId,
-      ),
-    };
+    const updatedState = deleteCommentFromState(state, commentId, threadId);
```
Comment (pranavkanade) [inline app/client/src/reducers/uiReducers/commentsReducer/commentsReducer.ts:149]: Should we reuse `handleDeleteCommentEvent`? Can also merge `DELETE_COMMENT_SUCCESS` and `DELETE_COMMENT_EVENT`

**Review by pranavkanade — DISMISSED** [review]
minor comments, rest looks good. Approving for test

File: app/client/src/reducers/uiReducers/commentsReducer/commentsReducer.ts (hunk 2 above)
Comment (rishabhsaxena) [reply app/client/src/reducers/uiReducers/commentsReducer/commentsReducer.ts:218]: Yeah actually that makes sense, have created an issue to merge handlers for all such events:
https://github.com/appsmithorg/appsmith/issues/6621

File: app/client/src/reducers/uiReducers/commentsReducer/commentsReducer.ts (hunk 3 above)
Comment (rishabhsaxena) [reply app/client/src/reducers/uiReducers/commentsReducer/commentsReducer.ts:150]: Created an issue for refactoring

**Review by pranavkanade — APPROVED** [review]
LGTM!

**Comment by rishabhsaxena** [comment]
Merging this since an unrelated and a known flaky test: `chart_spec`, is failing

**Pull request merged**